src/extraction/FStar.Extraction.ML.Code.fs

```diff
@@ -133,11 +133,13 @@ let infix_prim_ops = [
 ]
 
 (* -------------------------------------------------------------------- *)
-let prim_uni_ops = [
-    ("op_Negation", "not");
-    ("op_Minus", "~-");
-    ("op_Bang","Support.ST.read")
-]
+let prim_uni_ops = 
+    let op_minus = if Util.codegen_fsharp() 
```

Comment:
Does this happen late enough? (After options are parsed?)

Reply:
Probably not - I think this is the reason that I get `failiure: Option.get called on None` when building with OCaml. This works fine when building with F# though. 

What would you suggest as a fix? The unary negation issue only affects extraction, and extraction should be happening only after options are parsed. is `prim_uni_ops` in use somewhere before options are parsed?

Reply:
> extraction should be happening only after options are parsed. is prim_uni_ops in use somewhere before options are parsed?

It doesn't matter where it's used, if I'm reading this correctly: the call to `Util.codegen_fsharp()` happens right after the program is started, since this is a static, global variable — that is, it can happen before we even start running main.

The solution is to thunk this definition: to change it to `let prim_uni_ops () = …` and call the function where appropriate.